src/js/step.js

```diff
@@ -450,14 +409,13 @@ export class Step extends Evented {
 
     document.body.setAttribute('data-shepherd-step', this.id);
 
-    this.setupPopper();
-
     if (this.options.scrollTo) {
       setTimeout(() => {
         this.scrollTo();
       });
     }
 
+    this.tooltipElem.show();
```

Comment:
We should be setting up these tooltips when show is called, right? I think we need some setup code to replace `setupPopper` here. The idea is to create tooltips/poppers as needed, not to create them all at once, so they were being set up on show, and torn down when we cancel the tour, to minimize both the number of operations, and the amount of DOM we have lying around.

Reply:
See https://github.com/shipshapecode/shepherd/pull/261#discussion_r222499052. 

I'm fairly certain everything is happening as you described.

Reply:
If we don't already have some tests that cover this, we should write some tests to check that the tippy/popper for each step is not created until the step is shown, and that all `.tippy` `.shepherd` etc elements are gone after the tour is completed or cancelled. With those tests in place and passing, I'm fine with whatever here.

Reply:
@rwwagner90 Just updated this in the most recent commit.